Fix: How to check the image extension from Uri.

rustbasic opened the pull request:
Fix: How to check the image extension from Uri.

* Related #5324
* Related #5341

It addresses most of what was discussed in #5341, but I'm not sure if it addresses everything.

rustbasic committed:
Update image_loader.rs

rustbasic committed:
Update image_loader.rs

rustbasic committed:
Update image_loader.rs

rustbasic committed:
Update image_loader.rs

rustbasic committed:
Merge branch 'emilk:master' into patch145

xangelix commented:
Thank you for the PR and apologies again that my previous changes have caused issues for you, though I have concerns that this PR would still fail in some web URI scenarios.

Also, if this were to be the direction we go, you'd almost certainly want to do something like this to avoid having to keep extensions in sync with the image crate manually :)

```rs
    ImageFormat::all()
        .filter(ImageFormat::reading_enabled)
        .flat_map(ImageFormat::extensions_str)
        .any(|format_ext| uri.contains(*format_ext))
```

rustbasic committed:
Update image_loader.rs

rustbasic commented:
> Thank you for the PR to resolve an issue I introduced and apologies that my previous changes have caused issues for you, though I have concerns that this PR would still fail in some web URI scenarios.
> 
> Also, if this were to be the direction we go, you'd almost certainly want to do something like this to avoid having to keep extensions in sync with the image crate manually :)
> 
> ```rust
>     ImageFormat::all()
>         .filter(ImageFormat::reading_enabled)
>         .flat_map(ImageFormat::extensions_str)
>         .any(|format_ext| uri.contains(*format_ext))
> ```

I have modified this PR as you said.
If you want to make a PR, I will close this PR once you submit it.

rustbasic commented on crates/egui_extras/src/loaders/image_loader.rs:
```diff
@@ -19,16 +19,11 @@ impl ImageCrateLoader {
-        return true;
-    };
-
     // Uses only the enabled image crate features
```
In most cases it's fine, but I'm not sure if it's a good idea to remove this part.

rustbasic committed:
Update image_loader.rs

rustbasic committed:
Update image_loader.rs

rustbasic committed:
Merge branch 'emilk:master' into patch145

emilk commented:
Closed by https://github.com/emilk/egui/pull/5431
